src/wasm/wasm-type.cpp

```diff
@@ -124,11 +120,9 @@ void Type::init(const std::vector<Type>& types) {
     if (lookup()) {
       return;
     }
-    if (typeLists.size() >= (1 << ID_BITS)) {
-      WASM_UNREACHABLE("Too many types!");
-    }
-    id = typeLists.size();
-    typeLists.push_back(std::make_unique<std::vector<Type>>(types));
+    auto* vec = new std::vector<Type>(types);
```

Comment:
where is this deleted?

Reply:
NEVER! The canonical type vectors live for the length of the process, just like they did before.

Reply:
That sounds ok, but before it was making a unique ptr, so it was cleaned up, wasn't it?

Reply:
Yes, the unique_ptr was cleaned up, but only when global destructors run at the end of the program. I could make this a unique_ptr again but I'm not sure whether that has any value.

Reply:
It would avoid false positives with leak detection tools. I'm unsure myself if that is worth the cost.